### ISSUE TITLE
Segment with 3 points

### ISSUE THREAD

**Issue by yeldarby** [ultralytics/yolov5#6931]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I'm working on adding polygon support to Roboflow annotations so folks can benefit from [your copy-paste augmentation](https://github.com/ultralytics/yolov5/pull/3845) when training their custom datasets.

I'm looking at the YOLOv5 [segmentation dataloader as implemented here](https://github.com/ultralytics/yolov5/blob/master/utils/datasets.py#L909) and it looks like a polygon annotation requires at least 4 points to be considered a segment vs a box. I also saw @glenn-jocher mention [a minimum of 9 columns here](https://github.com/ultralytics/yolov5/pull/3524#issuecomment-856694841).

How should a triangle (7 columns, `class x1 y1 x2 y2 x3 y3`) be represented? Should I pad it with a 4th point that's a duplicate of the first?

**Comment by glenn-jocher** [comment]
@yeldarby oh hey good question! Yes, segmentation labels benefit from access to copy-paste augmentation, but also from smarter box cropping during 'normal' (scale, rotate, translate) augmentation, solving the issue below in https://github.com/ultralytics/yolov5/issues/2151.

![image](https://user-images.githubusercontent.com/26833433/157688768-b2f7ee30-9e0b-40a8-a954-32c785d24051.png)

You can see an example of segmentation labels vs box labels here:

- https://github.com/ultralytics/yolov5/releases/download/v1.0/coco128-segments.zip
- https://github.com/ultralytics/yolov5/releases/download/v1.0/coco128.zip

About how to handle a triangle, I didn't account for triangle shapes originally as I thought each segment would have at least 4 vertices. The logic implementing this is here, which will assume any row with > 8 columns to be a segment annotation instead of a box annotation. I think this should be reducible to > 6 though, which would then be able to accomodate triangle segments. I'll create a PR to test for any side effects of the change.
https://github.com/ultralytics/yolov5/blob/055e72af5b887832d5e7267ac9226c825d498cd2/utils/datasets.py#L907-L913

**Comment by yeldarby** [comment]
Awesome! I'll allow 3-vertex exports to YOLOv5 format on our end & revisit+revert if it turns out there are side-effects and that PR can't be merged.

**Comment by glenn-jocher** [comment]
@yeldarby good news 😃! Your original issue may now be fixed ✅ in PR #6938. This PR allows for 3-point segment labels containing 7 columns: (class xy1 xy2 xy3). Tests show no ill effects on normal and segment COCO128 training.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by Steinwang** [comment]
> Awesome! I'll allow 3-vertex exports to YOLOv5 format on our end & revisit+revert if it turns out there are side-effects and that PR can't be merged.

Hello,I'm using YOLOv5 for a detection and segmentation network, may I ask how you get segment in pbar? I'm in trouble with it,Thanks

**Comment by glenn-jocher** [comment]
@Steinwang what's pbar?
